File: 2st/Yeonwoo/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    x = 0
    while(True):
        next = []
        for i in arr:
            if i >= 50 and i % 2 == 0 :
                next.append(int(i/2))
            elif i < 50 and i % 2 == 1:
                next.append(i*2 + 1)
            else:
                next.append(i)
                
        if arr == next:
            return x
        else:
            x = x + 1
            arr = next

File: 3st/Juhee/조건에 맞게 수열 변환하기 2.java
class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        int count = 0;
        int[] a = new int[arr.length];
        while(count != arr.length) {
            count = 0;
            a = arr.clone();
            for(int i =0; i<arr.length; i++) {
            if(arr[i]%2 == 0 && arr[i] >= 50) {
                arr[i] /= 2;
            }else if(arr[i]%2 != 0 && arr[i] < 50) {
                arr[i] = arr[i]*2 +1;
            }
            if(arr[i] == a[i] ) count ++;
        }
            answer ++;
        }
        
        return answer-1;
    }
}